cirq-core/cirq/ops/boolean_hamiltonian_test.py

```diff
@@ -131,9 +131,7 @@ def test_gray_code_sorting(n_bits, expected_hs):
             x //= 2
         hs_template.append(tuple(sorted(h)))
 
-    for seed in range(10):
-        random.seed(seed)
-
+    for _ in range(10):
```

Comment:
This test _might_ have random behavior, if any other tests forget to restore the random state. I think we should try to avoid such coupling between tests.

Reply:
I'd prefer `random.shuffle` in this test to depend on CIRQ_TESTING_RANDOM_SEED.
In its initial form it would always produce the same output per constant seeds 0,..,9.
and it also reseeded RNG to 9 for any tests calling it later.